classes/Customer.php

```diff
@@ -773,6 +773,45 @@ public static function checkPassword($idCustomer, $passwordHash)
         return Cache::retrieve($cacheId);
     }
 
+    /**
+     * Validate that fields from Customer that are
+     * - required
+     * - boolean
+     * Must be true
+     *
+     * @return array of errors
+     *
+     * @throws PrestaShopException
+     */
+    public function validateCheckboxRequiredFields()
+    {
+        $this->cacheFieldsRequiredDatabase();
+        $errors = [];
+        $requiredFields = $this->getCachedFieldsRequiredDatabase();
+
+        foreach ($this->def['fields'] as $field => $fieldConfiguration) {
+            if (!in_array($field, $requiredFields)) {
+                continue;
+            }
+            if ($fieldConfiguration['type'] !== self::TYPE_BOOL) {
+                continue;
+            }
+
+            $value = Tools::getValue($field, null);
+
+
+            if ($value === false) {
```

Comment:
```suggestion
            if ($value !== true) {
```
What if the $field is null or 'blyat' ?

Reply:
If the field is null, then another check will take care of this. We are only looking for `false` values.
For `blyat` ... well I have no idea ! What are we supposed to do with a checkbox whose provided value is `blyat` ..? It could be a valid answer 😆